pkg/resource/statefulset.go

```diff
@@ -157,6 +157,7 @@ func (b StatefulSetBuilder) makePodTemplate() corev1.PodTemplateSpec {
 			TerminationGracePeriodSeconds: ptr.Int64(60),
 			Containers:                    b.MakeContainers(),
 			AutomountServiceAccountToken:  ptr.Bool(false),
+			ServiceAccountName:            "cockroach-operator-sa",
```

Comment:
```suggestion
			ServiceAccountName:            "cockroach-database-sa",
```

Reply:
This is a database and not the operator.  Please update it in the operator.yaml as well.